packages/shared_preferences/shared_preferences/lib/src/devtools_extension.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:developer' as developer;
+
+import 'package:flutter/foundation.dart';
+
+import '../shared_preferences.dart';
+
+const String _eventPrefix = 'shared_preferences:';
+
+/// A typedef for the post event function
+@visibleForTesting
+typedef PostEvent = void Function(
+  String eventKind,
+  Map<String, Object?> eventData,
+);
+
+/// A helper class that provides data to the devtool extension.
+///
+/// It is only visible for testing and eval.
+@visibleForTesting
+class DevtoolsExtension {
+  /// The default constructor for [DevtoolsExtension].
+  ///
+  /// Accepts an optional [PostEvent] that should only be overwritten when testing.
+  DevtoolsExtension([this._postEvent = developer.postEvent]);
+
+  final PostEvent _postEvent;
+
+  /// Requests all legacy and async keys and post an event with the result.
+  Future<void> requestAllKeys() async {
+    final SharedPreferences legacyPrefs = await SharedPreferences.getInstance();
+    final Set<String> legacyKeys = legacyPrefs.getKeys();
+    final Set<String> asyncKeys = await SharedPreferencesAsync().getKeys();
+
+    _postEvent('${_eventPrefix}all_keys', <String, List<String>>{
+      'asyncKeys': asyncKeys.toList(),
+      'legacyKeys': legacyKeys.toList(),
+    });
+  }
+
+  /// Requests the value for a given key and post and event with the result.
+  Future<void> requestValue(String key, bool legacy) async {
+    final Object? value;
+    if (legacy) {
+      final SharedPreferences legacyPrefs =
+          await SharedPreferences.getInstance();
+      value = legacyPrefs.get(key);
+    } else {
+      value = await SharedPreferencesAsync().getAll(allowList: <String>{
+        key
+      }).then((Map<String, Object?> map) => map.values.firstOrNull);
+    }
+
+    _postEvent('${_eventPrefix}value', <String, Object?>{
+      'value': value,
+      // It is safe to use `runtimeType` here. This code
+      // will only ever run in debug mode.
+      'kind': value.runtimeType.toString(),
+    });
+  }
+
+  /// Requests the a value change for the give key and post an empty event when finished;
```

Comment:
typos: 
- remove first "a"
- r/give/given
- r/post/posts
- period at end of sentence instead of semi colon